platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/DecryptMigrationManagerImpl.java

```diff
@@ -0,0 +1,159 @@
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DecryptMigrationManagerImpl.class);
+
+  private final MigrationReport report;
+
+  private final MigrationZipFile zip;
+
+  private final ZipOutputStream zipOutputStream;
+
+  private final Path exportFile;
+
+  private final Path decryptFile;
+
+  private boolean close = false;
+
+  @SuppressWarnings(
+      "squid:S2095" /* up to the caller of this method to close it; exception is thrown if the stream cannot be created in the first place */)
+  private static ZipOutputStream newZipOutputStreamFor(Path decryptFile) {
+    Validate.notNull(decryptFile, "invalid null decrypt file");
+    try {
+      return new ZipOutputStream(
+          new BufferedOutputStream(new FileOutputStream(decryptFile.toFile())));
+    } catch (SecurityException | FileNotFoundException e) {
+      throw new MigrationException(Messages.DECRYPT_FILE_CREATE_ERROR, decryptFile, e);
+    }
+  }
+
+  /**
+   * Creates a new migration manager for a decrypt operation.
+   *
+   * @param report the migration report where warnings and errors can be recorded
+   * @param zip the exported zip file
+   * @param decryptFile the file where to decrypt the exorted zip
+   * @throws MigrationException if a failure occurs while processing the zip file (the error will
+   *     not be recorded with the report)
+   * @throws IllegalArgumentException if <code>report</code> is <code>null</code> or if it is not
+   *     for a decrypt migration operation or if <code>zip</code> or <code>decryptFile</code> is
+   *     <code>null</code>
+   */
+  public DecryptMigrationManagerImpl(
+      MigrationReport report, MigrationZipFile zip, Path decryptFile) {
+    this(report, zip, decryptFile, DecryptMigrationManagerImpl.newZipOutputStreamFor(decryptFile));
+  }
+
+  @VisibleForTesting
+  DecryptMigrationManagerImpl(
+      MigrationReport report, MigrationZipFile zip, Path decryptFile, ZipOutputStream zos) {
+    Validate.notNull(report, "invalid null report");
+    Validate.notNull(zip, "invalid null zip file");
+    Validate.notNull(decryptFile, "invalid decrypt file");
+    Validate.isTrue(
+        report.getOperation() == MigrationOperation.DECRYPT, "invalid migration operation");
+    this.report = report;
+    this.exportFile = zip.getZipPath();
+    this.zip = zip;
+    this.decryptFile = decryptFile;
+    this.zipOutputStream = zos;
+  }
+
+  /**
+   * Proceeds with the decryp operation.
+   *
+   * @param productBranding the product branding being decrypted
+   * @param productVersion the product version being decrypted
+   * @throws IllegalArgumentException if <code>productBranding</code> or </code><code>productVersion
+   *     </code> is <code>null</code>
+   * @throws MigrationException to stop the decrypt operation
+   */
+  public void doDecrypt(String productBranding, String productVersion) {
+    Validate.notNull(productBranding, "invalid null product branding");
```

Comment:
❓ Do we accept blank branding an versions?

Reply:
The framework doesn't validate this info, it just reports it leaving its format to be whatever the system decides it should be.